python-lib/dku_deeplearning_image/config.py

```diff
@@ -0,0 +1,177 @@
+
+
+def add_retrain_recipe_config(dku_config, config):
+    dku_config.add_param(name='col_filename', value=config.get("col_filename"), required=True)
+    dku_config.add_param(name='col_label', value=config.get("col_label"), required=True)
+    dku_config.add_param(name='train_ratio', value=float(config.get("train_ratio")))
+    dku_config.add_param(
+        name='input_shape',
+        value=(int(config.get("image_height")), int(config.get("image_width")), 3))
+    dku_config.add_param(name='batch_size', value=int(config.get("batch_size")))
+    dku_config.add_param(name='model_pooling', value=config.get("model_pooling"))
+    dku_config.add_param(name='model_reg', value=config.get("model_reg"))
+    dku_config.add_param(name='model_dropout', value=float(config.get("model_dropout")))
+    dku_config.add_param(name='layer_to_retrain', value=config.get("layer_to_retrain"))
+    dku_config.add_param(name='layer_to_retrain_n', value=int(config.get("layer_to_retrain_n")), required=True)
+    dku_config.add_param(name='optimizer', value=config.get("model_optimizer"))
+    dku_config.add_param(name='learning_rate', value=config.get("model_learning_rate"))
+    dku_config.add_param(name='custom_params_opti', value=config.get("model_custom_params_opti"))
+    dku_config.add_param(name='nb_epochs', value=int(config.get("nb_epochs")))
+    dku_config.add_param(name='nb_steps_per_epoch', value=int(config.get("nb_steps_per_epoch")))
+    dku_config.add_param(name='nb_validation_steps', value=int(config.get("nb_validation_steps")))
+    dku_config.add_param(name='data_augmentation', value=config.get("data_augmentation"))
+    n_augmentation = int(config.get("n_augmentation")) if dku_config.data_augmentation else 0
+    dku_config.add_param(
+        name='n_augmentation',
+        value=n_augmentation,
+        checks=[
+            {
+                'type': 'custom',
+                'cond': not n_augmentation or (n_augmentation <= dku_config.dku_config.batch_size),
+                'err_msg': "The number of augmentations must be lower than the batch size. Aborting."
+            }
+        ]
+    )
+    dku_config.add_param(
+        name='custom_params_data_augment',
+        value=config.get("model_custom_params_data_augmentation", []))
+    dku_config.add_param(name='use_tensorboard', value=config.get("tensorboard"))
+    dku_config.add_param(name='random_seed', value=int(config.get("random_seed")))
+
+
+def add_api_deployer_config(dku_config, config, project):
+
+    dku_config.add_param(
+        name='model_folder_id',
+        value=config.get("model_folder_id"),
+        checks=[{
+            'type': 'in',
+            'op': [folder.get("id") for folder in project.list_managed_folders()],
+            'err_msg': "Folder ID {value} must be the id of a managed folder containing a model trained with the "
+                       "deeplearning-image plugin. The folder must belong to the project in which is executed the macro"
```

Comment:
nitpicking, `deep learning image v2` instead of just `deeplearning-image`

Reply:
Changed in ba0a631